exp/api/v1beta2/rosaroleconfig_types.go

```diff
@@ -0,0 +1,158 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	rosacontrolplanev1 "sigs.k8s.io/cluster-api-provider-aws/v2/controlplane/rosa/api/v1beta2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ROSARoleConfigSpec defines the desired state of ROSARoleConfig
+
+type ROSARoleConfigSpec struct {
+	AccountRoleConfig  AccountRoleConfig             `json:"accountRoleConfig"`
+	OperatorRoleConfig OperatorRoleConfig            `json:"operatorRoleConfig"`
+	OIDCConfig         OIDCConfig                    `json:"oidcConfig"`
+	IdentityRef        *infrav1.AWSIdentityReference `json:"identityRef,omitempty"`
+	Region             string                        `json:"region,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=rosaroleconfigs,scope=Namespaced,categories=cluster-api,shortName=rosarole
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// ROSARoleConfig is the Schema for the rosaroleconfigs API
+type ROSARoleConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ROSARoleConfigSpec   `json:"spec,omitempty"`
+	Status ROSARoleConfigStatus `json:"status,omitempty"`
+}
+
+type AccountRoleConfig struct {
+	// TODO: verify max len
+	// +kubebuilder:validation:MaxLength:=4
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix"`
+	// +optional
+	PermissionsBoundaryARN string `json:"permissionsBoundaryARN,omitempty"`
+	// +optional
+	Path string `json:"path,omitempty"`
+	// +kubebuilder:validation:Required
+	Version string `json:"version"`
+	// +optional
+	SharedVPCConfig SharedVPCConfig `json:"sharedVPCConfig,omitempty"`
+}
+
+type OperatorRoleConfig struct {
+	// +kubebuilder:validation:MaxLength:=4
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix"`
+	// +optional
+	PermissionsBoundaryARN string `json:"permissionsBoundaryARN,omitempty"`
+	// +kubebuilder:validation:Required
+	OIDCConfigID string `json:"oidcConfigId,omitempty"`
+	// +optional
+	SharedVPCConfig SharedVPCConfig `json:"sharedVPCConfig,omitempty"`
+}
+
+type SharedVPCConfig struct {
+	RouteRoleARN       string `json:"routeRoleARN,omitempty"`
+	VPCEndpointRoleARN string `json:"vpcEndpointRoleArn,omitempty"`
+}
+
+type OIDCConfig struct {
+	ManagedOIDC           bool                                      `json:"managedOIDC"`
+	ExternalAuthProviders []rosacontrolplanev1.ExternalAuthProvider `json:"externalAuthProviders,omitempty"`
```

Comment:
You need to set the field desc as the controlplane like [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/controlplane/rosa/api/v1beta2/rosacontrolplane_types.go#L139)